src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LENGTH = 10;
+    private int size;
+    private Object[] array;
+
+    public ArrayList() {
+        this.array = new Object[DEFAULT_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        if (array.length == size) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index: " + index);
+        }
+        if (array.length == size) {
+            grow();
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
```

Comment:
Магічні числа можна замінити на константи
```suggestion
        System.arraycopy(array, index, array, index + NEXT_ELEMENT_FROM_INDEX, size - index);
```

Reply:
done